### PR TITLE
COL-419 Feature/doi registration poc

### PR DESCRIPTION
**Pull request by a-luz**
## Purpose

Added tables and functions to query tables for DOI.
Integration with Zenodos API for creating and retrieving a DOI and uploading files to it.

This is a proof of concept created to try out Zenodo's API.

## Related Issues

Closes COL-419

## Submission Checklist

- [x] Included Jira issue in the PR title (e.g. `COL-### Did something here`)
- [x] Code passes linter rules for each file you updated. To lint all files at once, run `npm run lint`. To just lint one specific file run `npx quick-lint-js src/js/<file-you-changed> --snarky`.
- [x] No new reflection warnings (`clojure -M:check-reflection`)

### CODE REVIEW

File: src/clj/collect_earth_online/db/doi.clj [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  (map (fn [user]
+         {:name        (:email user)
+          :type        "DataCurator"
+          :affiliation institution-name})
+       users))
+
+(defn create-zenodo-deposition
+  [institution-name
+   project-name
+   user users-assigned
```
Comment (sifbiri) [inline src/clj/collect_earth_online/db/doi.clj:35]: Format the arguments (user, user-assigned) by placing them on separate lines, similar to the other

File: src/clj/collect_earth_online/db/doi.clj [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            [triangulum.type-conversion :as tc]
+            [collect-earth-online.generators.external-file :refer [create-shape-files]])
+  (:import java.time.format.DateTimeFormatter
+           java.time.LocalDateTime))
+
+(def base-url (:url (get-config :zenodo)))
+
+(defn req-headers
+  []
+  (let [auth-token (:api-key (get-config :zenodo))]
```
Comment (sifbiri) [inline src/clj/collect_earth_online/db/doi.clj:16]: The structure is likely to undergo some modifications when transitioning to the new Triangulum version, which will involve placing it under the :product-ns namespace.

Comment (sifbiri) [reply src/clj/collect_earth_online/db/doi.clj:16]: The structure is likely to undergo some modifications when transitioning to the new Triangulum version, which will involve placing it under the :product-ns (ceo) namespace.

File: src/clj/collect_earth_online/db/doi.clj (hunk 1 above)
Comment (sifbiri) [inline src/clj/collect_earth_online/db/doi.clj:35]: I think you can put user and users-assigned in new line

File: src/clj/collect_earth_online/db/doi.clj (hunk 2 above)
Comment (a-luz) [reply src/clj/collect_earth_online/db/doi.clj:16]: The transition to the new Triangulum version will still take a while. I'll start work on that probably in the last sprint of this month

File: src/clj/collect_earth_online/generators/external_file.clj [hunk 3]
```diff
@@ -241,3 +250,31 @@
...
+
+(defn pgsql2shp-string
+  [db-config file-name query]
+  (let [db-name (:dbname db-config)
+        db-user (:user db-config)]
+    (str "pgsql2shp -f " file-name " -u " db-user " " db-name " " query)))
+
+(defn export-table-to-file
+  [folder-name project-id table-name db-config]
+  (sh-wrapper-quoted folder-name {}
```
[the remark this answers is not in the document]
Comment (a-luz) [reply src/clj/collect_earth_online/generators/external_file.clj:262]: By the end of the month, I'll be migrating to the new Triangulum version. I'll make sure to use its shell functions

**Pull request merged**